attractions/snackbar/snackbar.svelte

```diff
@@ -6,14 +6,47 @@
 
   let _class = null;
   export { _class as class };
+  /**
+   * A class string to add to the label of the snackbar.
+   * @type {string}
+   */
   export let textClass = null;
+  /**
+   * A class string to add to the action button of the snackbar.
+   * @type {string}
+   */
   export let buttonClass = null;
 
+  /**
+   * The text to show on the snackbar.
+   * @type {string}
+   */
   export let text;
+  /**
+   * The action for the button on the snackbar. If this is null, the button is not rendered. Otherwise it has text as a label and calls callback on click.
+   * @type {{text: string; callback: () => void}}
+   */
   export let action = null;
+  /**
+   * Whether to call the `closeCallback` when an action button is pressed, after its callback.
+   * @type {boolean}
+   */
   export let closeOnAction = true;
+  /**
+   * The callback to call to close the snackbar in the `SnackbarContainer`.
+   * Passed internally by the `SnackbarContainer` when using the `showSnackbar` function.
+   * @type {() => void}
+   */
   export let closeCallback;
+  /**
+   * The transition to use to animate the in and out of the snackbar.
+   * @type {() => import('svelte/transition').TransitionConfig}
+   */
   export let transition = fly;
+  /**
+   * The options to use with the given transition.
+   * @type {Record<string, any>}
```

Comment:
I don't think we should be imposing any limitations here, I'd say have it be an `object`

Reply:
`object` means any non-primitive, which includes arrays and basically anything constructed using `new`. `Record` is the TypeScript way of saying "the object we are familiar with that we can represent as a literal `{ key: value }`"

Reply:
Yeah, but there can be a transition that takes an array as an option. This is entirely up to the transition object, therefore, this constraint might cause errors.

Reply:
And can't there be a transition that takes a number (or any primitive value)? Isn't `any` more appropriate then?

Reply:
True, make it `any` then